android/src/main/java/io/radar/capacitor/RadarPlugin.java

```diff
@@ -208,29 +242,121 @@ public void getLocationPermissionsStatus(PluginCall call) {
         call.resolve(ret);
     }
 
+    @PluginMethod
+    public void getBeaconsPermissionStatus(PluginCall call) {
+        String beaconsAuthorization = "NOT_DETERMINED";
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            PermissionState beaconPermissions;
+            String alias = "beacons";
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                alias = "beaconsAndroid12";
+            }
+            beaconPermissions = getPermissionState(alias);
+            if (PermissionState.GRANTED == beaconPermissions) {
+                beaconsAuthorization = "DENIED";
+            } else {
+                beaconsAuthorization = "GRANTED";
+            }
+        }
+        JSObject ret = new JSObject();
+        ret.put("status", beaconsAuthorization);
+        call.resolve(ret);
+    }
+
     @PluginMethod
     public void requestLocationPermissions(PluginCall call) {
         if (!call.hasOption("background")) {
             call.reject("background is required");
             return;
         }
-        //noinspection ConstantConditions
-        boolean background = call.getBoolean("background", false);
+        Boolean background = call.getBoolean("background", false);
 
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-            int requestCode = 0;
-            if (background && Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-                pluginRequestPermissions(new String[]{
-                        Manifest.permission.ACCESS_FINE_LOCATION,
-                        Manifest.permission.ACCESS_BACKGROUND_LOCATION
-                }, requestCode);
+            PermissionState backgroundState = PermissionState.GRANTED;
+            PermissionState locationPermission = getPermissionState("location");
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                if (Boolean.TRUE == background) {
```

Comment:
can this condition be `if (background)`?

Reply:
Technically it could, but the lint warnings identify a potential NPE if we do that.